src/Controller/UserController.php

```diff
@@ -265,39 +265,28 @@ public function enableTwoFactorAuthAction(Request $req, #[VarName('name')] User
             throw $this->createAccessDeniedException('You cannot change this user\'s two-factor authentication settings');
         }
 
-        $enableRequest = new EnableTwoFactorRequest();
-        $form = $this->createForm(EnableTwoFactorAuthType::class, $enableRequest)
-            ->handleRequest($req);
-
-        $secret = (string) $req->getSession()->get('2fa_secret');
-        if (!$form->isSubmitted() || '' === $secret) {
-            $secret = $authenticator->generateSecret();
-            $req->getSession()->set('2fa_secret', $secret);
-        }
-
+        $secret = (string) $req->getSession()->get('2fa_secret', $authenticator->generateSecret());
```

Comment:
this is now generating a new secret (that will be ignored) even when the session already has one. If the generation takes time, it might be a bad idea.

I suggest doing `$secret = (string) $req->getSession()->get('2fa_secret') ?? $authenticator->generateSecret();` instead